cpp/src/unary/cast_ops.cu

```diff
@@ -94,24 +95,72 @@ struct unary_cast {
+                                       cudf::is_numeric<TargetT>())>* = nullptr>
+  CUDA_DEVICE_CALLABLE TargetT operator()(DeviceT const element)
+  {
+    auto const fp = SourceT{numeric::scaled_integer<DeviceT>{element, scale}};
+    return static_cast<TargetT>(fp);
+  }
+
+  template <typename SourceT                                              = _SourceT,
+            typename TargetT                                              = _TargetT,
+            typename std::enable_if_t<(cudf::is_numeric<_SourceT>() &&
+                                       cudf::is_fixed_point<TargetT>())>* = nullptr>
+  CUDA_DEVICE_CALLABLE DeviceT operator()(SourceT const element)
+  {
+    auto const fp = TargetT{element, scale};
+    return numeric::scaled_integer<DeviceT>{fp}.value;
+  }
+};
+
+template <typename From, typename To>
+constexpr inline auto is_supported_non_fixed_point_cast()
+{
+  return cudf::is_fixed_width<To>() &&
+         // Disallow fixed_point here (requires different specialization)
+         !(cudf::is_fixed_point<From>() || cudf::is_fixed_point<To>()) &&
+         // Disallow conversions between timestamps and numeric
+         !(cudf::is_timestamp<From>() && is_numeric<To>()) &&
+         !(cudf::is_timestamp<To>() && is_numeric<From>());
+}
+
+template <typename From, typename To>
+constexpr inline auto is_supported_fixed_point_cast()
+{
+  return (cudf::is_fixed_point<From>() && cudf::is_numeric<To>()) ||
+         (cudf::is_numeric<From>() && cudf::is_fixed_point<To>());
+}
+
+template <typename From, typename To>
+constexpr inline auto is_supported_cast()
+{
+  return is_supported_non_fixed_point_cast<From, To>() || is_supported_fixed_point_cast<From, To>();
+}
+
 template <typename SourceT>
 struct dispatch_unary_cast_to {
   column_view input;
 
   dispatch_unary_cast_to(column_view inp) : input(inp) {}
 
-  // Disallow conversions between timestamps and numeric
   template <
     typename TargetT,
-    typename std::enable_if_t<cudf::is_fixed_width<TargetT>() && !cudf::is_fixed_point<TargetT>() &&
-                              !(cudf::is_timestamp<SourceT>() && is_numeric<TargetT>()) &&
-                              !(cudf::is_timestamp<TargetT>() && is_numeric<SourceT>())>* = nullptr>
+    typename std::enable_if_t<is_supported_non_fixed_point_cast<SourceT, TargetT>()>* = nullptr>
   std::unique_ptr<column> operator()(data_type type,
                                      rmm::mr::device_memory_resource* mr,
                                      cudaStream_t stream)
   {
-    auto size   = input.size();
-    auto output = std::make_unique<column>(type,
+    auto const size = input.size();
+    auto output     = std::make_unique<column>(type,
                                            size,
                                            rmm::device_buffer{size * cudf::size_of(type), 0, mr},
                                            copy_bitmask(input, 0, mr),
```

Comment:
Should be passing the `stream` parameter in these two lines instead of 0.

Reply:
Done.